resolver/resolver.go

```diff
@@ -69,15 +78,47 @@
 }
 
 func (resolver *Resolver) QueryFees(ctx context.Context, id interface{}, params *jsonrpc.ParamsQueryFees, req *http.Request) jsonrpc.Response {
-	return resolver.handleMessage(ctx, id, jsonrpc.MethodQueryStat, *params, req, false)
+	return resolver.handleMessage(ctx, id, jsonrpc.MethodQueryFees, *params, req, false)
 }
 
-func (resolver *Resolver) QueryTxs(ctx context.Context, id interface{}, params *jsonrpc.ParamsQueryTxs, r *http.Request) jsonrpc.Response {
-	// TODO: Implement the queryTx method.
-	return jsonrpc.NewResponse(id, nil, &jsonrpc.Error{
-		Code:    jsonrpc.ErrorCodeMethodNotFound,
-		Message: "unsupported method",
-	})
+func (resolver *Resolver) QueryTxs(ctx context.Context, id interface{}, params *jsonrpc.ParamsQueryTxs, req *http.Request) jsonrpc.Response {
+	var tag string
+	if params.Tags != nil {
+		if len(*params.Tags) > resolver.serverOptions.MaxTags {
+			jsonErr := jsonrpc.NewError(jsonrpc.ErrorCodeInvalidParams, fmt.Sprintf("maximum number of tags is %d", resolver.serverOptions.MaxTags), nil)
+			return jsonrpc.NewResponse(id, nil, &jsonErr)
+		}
+		if len(*params.Tags) > 0 {
+			// Currently we only support a maximum of one tag, but this can be
+			// extended in the future.
+			tag = hex.EncodeToString((*params.Tags)[0][:])
+		}
+	}
+
+	var page uint64
+	if params.Page != nil {
+		page = params.Page.Int.Uint64()
+	}
```

Comment:
Probably better to saturate and clamp this value, instead of returning an error. If the pageSize exceeds the maximum, then just we just set the pageSize to be the maximum.